src/http/server/handlers/deflate_handler.cr

```diff
@@ -0,0 +1,33 @@
+require "zlib"
+
+class HTTP::DeflateHandler < HTTP::Handler
+  DEFLATE_TYPES = %w(text/html text/plain text/xml text/css text/javascript application/javascript)
```

Comment:
@ysbaddaden @luislavena done. I didn't find documentation of the identity. So I add only the no-transform and the bodyless responses. 

HTTP::Client should handle compression out of the box [as in ruby stdlib](http://ruby-doc.org/stdlib-2.1.1/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-Compression) ... so this might not be the last PR in the deflate series... Do you agree?